py_swf/clients/workflow.py

```diff
@@ -65,3 +66,177 @@ def terminate_workflow(self, workflow_id, reason):
+            oldest_close_date=None,
+            latest_close_date=None,
+            workflow_name=None,
+            version=None,
+            tag=None,
+            workflow_id=None,
+            close_status=None,
+    ):
+        """
+        Count the number of closed workflows for a domain. You can pass in filtering criteria.
+        The results are best effort and may not exactly reflect recent updates and changes.
+
+        Passthrough to :meth:~SWF.Client.count_closed_workflow_executions``
+
+        startTimeFilter and closeTimeFilter are mutually exclusive. You MUST specify one of these in a request but not both.
+        closeStatusFilter , executionFilter , typeFilter and tagFilter are mutually exclusive. You can specify at most
+            one of these in a request.
+
+        :param oldest_start_date: datetime. Specifies the oldest start date and time to return.
+        :param latest_start_date: datetime. Specifies the latest start date and time to return.
+        :param oldest_close_date: datetime. Specifies the oldest close date and time to return.
+        :param latest_close_date: datetime. Specifies the latest close date and time to return.
+        :param workflow_name: string. Required for typeFilter. Specifies the type of the workflow executions to be counted.
+        :param version: string. Optional for typeFilter. Version of the workflow type.
+        :param tag: string. Required for tagFilter. Specifies the tag that must be associated with the execution for it
+            to meet the filter criteria.
+        :param workflow_id: string. Required for executionFilter. The workflowId to pass of match the criteria of this filter.
+        :param close_status: string.
+            valid status are ('COMPLETED', 'FAILED', 'CANCELED', 'TERMINATED', 'CONTINUED_AS_NEW', 'TIMED_OUT')
+            This filter has an affect only if executionStatus is specified as CLOSED
+        :return: total number of closed workflows that meet the filter criteria
+        """
+        time_filter_dict = _build_time_filter_dict(
+            oldest_start_date=oldest_start_date,
+            latest_start_date=latest_start_date,
+            oldest_close_date=oldest_close_date,
+            latest_close_date=latest_close_date,
+        )
+
+        count_closed_workflow_within_time_range = partial(
+            self.boto_client.count_closed_workflow_executions,
+            domain=self.workflow_client_config.domain,
+            **time_filter_dict
+        )
+
+        workflow_filter_dict = _build_workflow_filter_dict(
+            workflow_name=workflow_name,
+            version=version,
+            tag=tag,
+            workflow_id=workflow_id,
+            close_status=close_status,
+        )
+
+        return count_closed_workflow_within_time_range(**workflow_filter_dict)
+
+
+def _build_time_filter_dict(oldest_start_date=None, latest_start_date=None, oldest_close_date=None, latest_close_date=None):
```

Comment:
If that is what you are worried about, I will try building both startTimeFilter dict and closeTimeFilter and put them in the result. In this case if user passes in both start date and close date then boto3 will get them both. This will cause exceptions and prevent silent errors.